src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java

```diff
@@ -0,0 +1,272 @@
+import org.springframework.stereotype.Component;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
+@Component
+public class GSMATransferDef {
+    public static String gsmaTransferBody;
+    public static String loanProductBody;
+    public static String amsName;
+    public static int amount;
+    public static String accountId;
+    public static String acccountHoldingInstitutionId;
+
+    public static String gsmaTransactionResponse;
+    public static String responseLoanProduct;
+    public static String tenant;
+    public static String currentDate;
+    protected static String loanAccountBody;
+    protected static String responseLoanAccount;
+    protected static String loanApproveBody;
+    protected static String responseLoanApprove;
+    protected static String loanRepaymentBody;
+    protected static String responseLoanRepayment;
+    protected static String savingsProductBody;
+    protected static String responseSavingsProduct;
+    protected static String savingsApproveBody;
+    protected static String responseSavingsApprove;
+    protected static String savingsAccountBody;
+    protected static String responseSavingsAccount;
+    protected static String savingsActivateBody;
+    protected static String responseSavingsActivate;
+    protected static String savingsDepositAccountBody;
+    protected static String responseSavingsDepositAccount;
+    protected static String createPayerClientBody;
+    protected static String responsePayerClient;
+    protected static String payerClientBody;
+    protected static String loanDisburseBody;
+    protected static String responseLoanDisburse;
+    protected static String gsmaTransfer;
+
+    @Autowired
+    static ObjectMapper objectMapper;
+
+    Logger logger = LoggerFactory.getLogger(this.getClass());
+
+
+    public static void setAccountId(String acId) {
+        accountId = acId;
+    }
+
+    public static void setHeadersMifos(String ams, String aHIId, int amnt) {
+        amsName = ams;
+        acccountHoldingInstitutionId = aHIId;
+        amount = amnt;
+    }
+
+    protected static void setTenant(String tenantName) {
+        tenant = tenantName;
+    }
+
+    protected static void setcurrentDate(String date) {
+        currentDate = date;
+    }
+
+    protected static RequestSpecification setHeadersLoan(RequestSpecification requestSpec) {
+        requestSpec.header("Fineract-Platform-TenantId", tenant);
+        requestSpec.header("Accept", "application/json, text/plain, */*");
+        requestSpec.header("Accept-Language", "en-GB,en-US;q=0.9,en;q=0.8");
+        requestSpec.header("Accept-Encoding", "gzip, deflate, br");
+        requestSpec.header("Authorization", "Basic bWlmb3M6cGFzc3dvcmQ=");
+        requestSpec.header("Origin", "https://65.0.42.17:8443");
+        requestSpec.header("Referer", "https://65.0.42.17:8443/");
+        requestSpec.header("Sec-Fetch-Dest", "empty");
```

Comment:
all these headers are not mandatory

Reply:
keep the minimal headers, teannt, auth and content type only.  also read tenant and auth from config

Reply:
Tenant is read from feature file.

Reply:
The origin and refferar IP address are not necessary